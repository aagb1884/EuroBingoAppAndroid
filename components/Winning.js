import { StyleSheet, View, Pressable, Text } from 'react-native';
import { useEffect, useState } from 'react';
import { Audio } from 'expo-av';



export default function Winning({ isBingoButtonActive, setShowGif}) {
    const [sound, setSound] = useState();
    

    async function playSound() {
        console.log('Loading Sound');
        const { sound } = await Audio.Sound.createAsync( require('../assets/euphoria_excerpt.mp3')
        );
        setSound(sound);
    
        console.log('Playing Sound');
        await sound.playAsync();
        setShowGif(true);

        setTimeout(() => {
            setShowGif(false);
          }, 7300);
        };
      
    
      useEffect(() => {
        return sound
          ? () => {
              console.log('Unloading Sound');
              sound.unloadAsync();
            }
          : undefined;
      }, [sound]);

  

      return (
        <View 
        
        style={isBingoButtonActive? styles.buttonContainerActive: styles.buttonContainer}>
          
            <Pressable 
            disabled={!isBingoButtonActive}
            onPress={playSound}
            style={[isBingoButtonActive ? styles.button : styles.disabled]}
                       >
              <Text style={[isBingoButtonActive ? styles.activeText : styles.buttonLabel]}>BINGO</Text>
            </Pressable>
          
        </View>
      );
    }



      const styles = StyleSheet.create({
        buttonContainer: {
          width: 340,
          height: 68,
          marginHorizontal: 20,
          alignItems: 'center',
          justifyContent: 'center',
          padding: 3,
          borderRadius: 18,
        },
        buttonContainerActive: {
          width: 320,
          height: 68,
          marginHorizontal: 20,
          alignItems: 'center',
          justifyContent: 'center',
          padding: 3,
          borderWidth: 4,
          borderRadius: 18,
        },
        button: {
          borderRadius: 10,
          width: '100%',
          height: '100%',
          alignItems: 'center',
          justifyContent: 'center',
          flexDirection: 'row',
          backgroundColor: 'yellow', 
        },
        disabled: {
          borderRadius: 10,
          width: '100%',
          height: '100%',
          alignItems: 'center',
          justifyContent: 'center',
          flexDirection: 'row',
          backgroundColor: '#8C8C8C', 
          color: '#fff', 
          opacity: 0.5,
          },
         buttonLabel: {
          color: 'black',
          fontSize: 20,
          fontWeight: "bold",
          textAlign: 'center',
         },
          activeText: {
            color: 'black', 
            fontSize: 20,
            fontWeight: "bold",
            textAlign: "center"
       },
     });
      