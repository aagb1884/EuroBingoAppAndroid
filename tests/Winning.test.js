import React from 'react';
import renderer from 'react-test-renderer';

import Winning from '../components/Winning';

describe('<Winning />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Winning />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
});
