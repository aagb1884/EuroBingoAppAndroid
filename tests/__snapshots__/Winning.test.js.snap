// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Winning /> renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
    }
  }
>
  <View
    collapsable={false}
    style={
      {
        "alignItems": "center",
        "height": 68,
        "justifyContent": "center",
        "marginBottom": 10,
        "opacity": 0,
        "width": 340,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": true,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "yellow",
          "borderRadius": 10,
          "flexDirection": "row",
          "height": "100%",
          "justifyContent": "center",
          "opacity": 1,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          {
            "color": "black",
            "fontSize": 20,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        BINGO
      </Text>
    </View>
  </View>
  <View
    style={
      [
        {
          "alignItems": "center",
          "flexDirection": "row",
          "marginTop": 10,
          "transform": [
            {
              "scaleX": 1.2,
            },
            {
              "scaleY": 1.2,
            },
          ],
        },
        {
          "backgroundColor": "white",
          "color": "black",
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "marginRight": 10,
          },
          {
            "backgroundColor": "white",
            "color": "black",
          },
        ]
      }
    >
      Sound On/Off:
    </Text>
    <RCTSwitch
      accessibilityRole="switch"
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "height": 31,
          "width": 51,
        }
      }
      value={true}
    />
  </View>
</View>
`;
